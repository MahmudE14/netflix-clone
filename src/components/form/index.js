import { Container, Base, Error, Title, Text, TextSmall, Link, Input, Submit } from './styles/form'

export default function Form({ children, ...restProps }) {
    return (
        <Container {...restProps}>{children}</Container>
    )
};

Form.Base = function FormContainer({ children, ...restProps }) {
    return <Base { ...restProps }>{children}</Base>
}

Form.Error = function FormContainer({ children, ...restProps }) {
    return <Error { ...restProps }>{children}</Error>
}

Form.Title = function FormContainer({ children, ...restProps }) {
    return <Title { ...restProps }>{children}</Title>
}

Form.Text = function FormContainer({ children, ...restProps }) {
    return <Text { ...restProps }>{children}</Text>
}

Form.TextSmall = function FormContainer({ children, ...restProps }) {
    return <TextSmall { ...restProps }>{children}</TextSmall>
}

Form.Link = function FormContainer({ children, ...restProps }) {
    return <Link { ...restProps }>{children}</Link>
}

Form.Input = function FormContainer({ children, ...restProps }) {
    return <Input { ...restProps }>{children}</Input>
}

Form.Submit = function FormContainer({ children, ...restProps }) {
    return <Submit { ...restProps }>{children}</Submit>
}
