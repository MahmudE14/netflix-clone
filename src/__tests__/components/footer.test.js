import { render, screen } from "@testing-library/react";
import { Footer } from "../../components";

describe("<Footer />", () => {
  it("renders the <Footer /> with populated data", () => {
    const { container, getByText } = render(
      <Footer>
        <Footer.Title>Question? Contact Us.</Footer.Title>
        <Footer.Break />
        <Footer.Row>
          <Footer.Column>
            <Footer.Link href="#0">FAQ</Footer.Link>
            <Footer.Link href="#0">Investor Relations</Footer.Link>
            <Footer.Link href="#0">Ways to Watch</Footer.Link>
            <Footer.Link href="#0">Corporate Information</Footer.Link>
            <Footer.Link href="#0">Netflix Originals</Footer.Link>
          </Footer.Column>
        </Footer.Row>
        <Footer.Break />
        <Footer.Text>Netflix United Kingdom</Footer.Text>
      </Footer>
    );

    expect(screen.getByText('Question? Contact Us.')).toBeTruthy()
    expect(screen.getByText('FAQ')).toBeTruthy()
    expect(screen.getByText('Investor Relations')).toBeTruthy()
    expect(screen.getByText('Ways to Watch')).toBeTruthy()
    expect(screen.getByText('Corporate Information')).toBeTruthy()
    expect(screen.getByText('Netflix Originals')).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  });
});
