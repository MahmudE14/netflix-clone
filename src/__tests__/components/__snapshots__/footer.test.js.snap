// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the <Footer /> with populated data 1`] = `
<div
  class="footer__Container-sc-eh6b4z-0 dEUDCY"
>
  <p
    class="footer__Title-sc-eh6b4z-4 dLPDHa"
  >
    Question? Contact Us.
  </p>
  <div
    class="footer__Break-sc-eh6b4z-6 jiQmMZ"
  />
  <div
    class="footer__Row-sc-eh6b4z-2 cwnnLG"
  >
    <div
      class="footer__Column-sc-eh6b4z-1 dIxCTP"
    >
      <a
        class="footer__Link-sc-eh6b4z-3 fMtVCE"
        href="#0"
      >
        FAQ
      </a>
      <a
        class="footer__Link-sc-eh6b4z-3 fMtVCE"
        href="#0"
      >
        Investor Relations
      </a>
      <a
        class="footer__Link-sc-eh6b4z-3 fMtVCE"
        href="#0"
      >
        Ways to Watch
      </a>
      <a
        class="footer__Link-sc-eh6b4z-3 fMtVCE"
        href="#0"
      >
        Corporate Information
      </a>
      <a
        class="footer__Link-sc-eh6b4z-3 fMtVCE"
        href="#0"
      >
        Netflix Originals
      </a>
    </div>
  </div>
  <div
    class="footer__Break-sc-eh6b4z-6 jiQmMZ"
  />
  <div
    class="footer__Text-sc-eh6b4z-5 dsycrP"
  >
    Netflix United Kingdom
  </div>
</div>
`;
